painttiming.bs

```diff
@@ -62,9 +87,38 @@ Formally, we consider the user agent to have "rendered" a document when it has p
 
     NOTE: The rendering pipeline is very complex, and the timestamp should be the latest timestamp the user agent is able to note in this pipeline (best effort). Typically the time at which the frame is submitted to the OS for display is recommended for this API.
 
+An [=element=] is <dfn export>contentful</dfn> when one or more of the following apply:
+* The element has a [=text node=] child, with at least one character, excluding [=document white space characters=].
+* The element is a [=replaced element=] representing a [=loaded=] image.
+* The element's [=used=] style contains a [=loaded=] [=background-image=].
+* The element is a [=canvas=] with its [=context mode=] set to any value other than <code>none</code>.
+* The element is an [=svg element with rendered descendants=].
+* The element is an [=originating element=] for a [=paintable pseudo-element=] that represents a [=loaded=] image or text with at least one character excluding [=document white space characters=].
+
+An [=element=] is <dfn>paintable</dfn> when all of the following apply:
+* The element's [=used=] [=visibility=] is <code>visible</code>.
+* The element and all of its ancestors' [=used=] [=opacity=] is greater than zero.
+
+    NOTE: there could be cases where a <code>paintable</code> [=element=] would not be visible to the user, for example in the case of text that has the same color as its background.
+    Those elements would still considered as paintable for the purpose of computing [=first contentful paint=].    
+
+* The element's [=paintable bounding rect=] has positive right, bottom, width and height values. 
+    
+    NOTE: This covers the cases where the element is scaled to zero size, has <code>display: none</code>, or <code>display: contents</code> where the contents resolve to an empty rect.
+
+    NOTE: As a general rule, an [=element=] is paintable if it is within the viewport, or can potentially be in the viewport as a result of scrolling or zooming.
+
+A [=generated content pseudo-element=] is a <dfn>paintable pseudo-element</dfn> when all of the following apply:
```

Comment:
nit: this is used before it's defined, so perhaps we can move this definition to be above.